src/Microsoft.ML.Data/DebuggerExtensions.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Transforms;
+
+namespace Microsoft.ML
+{
+    /// <summary>
+    /// Static extensions for data preview.
+    /// </summary>
+    public static class DebuggerExtensions
+    {
+        /// <summary>
+        /// Extract a 'head' of the data view in a view that is convenient to debug.
+        /// </summary>
+        /// <param name="data">The data view to preview</param>
+        /// <param name="maxRows">Maximum number of rows to pull</param>
+        public static DataDebuggerPreview Preview(this IDataView data, int maxRows = DataDebuggerPreview.Defaults.MaxRows)
+            => new DataDebuggerPreview(data, maxRows);
+
+        /// <summary>
+        /// Preview an effect of the <paramref name="transformer"/> on a given <paramref name="data"/>.
+        /// </summary>
+        /// <param name="transformer">The transformer which effect we are previewing</param>
+        /// <param name="data">The data view to use for preview</param>
+        /// <param name="maxRows">Maximum number of rows to pull</param>
+        public static DataDebuggerPreview Preview(this ITransformer transformer, IDataView data, int maxRows = DataDebuggerPreview.Defaults.MaxRows)
+            => new DataDebuggerPreview(transformer.Transform(data), maxRows);
```

Comment:
Isn't this going to transform all the data, and then just show `maxRows` of them? Is that the behavior we want? I see below we are using a `SkipTakeFilter`.

Reply:
IDataView is lazy, so the `Transform()` itself doesn't have cost.

---
In reply to: [230913125](https://github.com/dotnet/machinelearning/pull/1518#discussion_r230913125) [](ancestors = 230913125)